pkg/upgrade/ready.go

```diff
@@ -85,22 +85,30 @@ var daemonsetWhitelist = []struct {
 	},
 }
 
-func WaitForNodes(ctx context.Context, cs *api.OpenShiftManagedCluster, kc *kubernetes.Clientset) error {
-	config := api.PluginConfig{AcceptLanguages: []string{"en-us"}}
-	authorizer, err := azureclient.NewAuthorizerFromContext(ctx)
-	if err != nil {
-		return err
+func (u *simpleUpgrader) waitForNodes(ctx context.Context, cs *api.OpenShiftManagedCluster) error {
+	var err error
+	if u.kubeclient == nil {
+		u.kubeclient, err = managedcluster.ClientsetFromV1ConfigAndWait(ctx, cs.Config.AdminKubeconfig)
```

Comment:
Waiting for the api server here seems a bit redundant, we should have healthchecking code for the api server elsewhere already, no?

Reply:
This is probably fine.